cassiopeia/solver/graph_utilities.py

```diff
@@ -194,8 +195,6 @@ def construct_similarity_graph(
         A similarity graph constructed over the sample set
     """
     G = nx.Graph()
-    if not samples:
-        samples = range(cm.shape[0])
     for i in samples:
```

Comment:
In the function definition, you have samples as possibly being an empty list. Let's make this a mandatory argument. But if you don't want to, then technically the typing should be wrapped in `Optional`